dclab/rtdc_dataset/plugins/plugin_feature.py

```diff
@@ -0,0 +1,66 @@
+"""
+.. versionadded:: 0.34.0
+"""
+
+import pathlib
+import importlib
+import sys
+
+from ..ancillaries import AncillaryFeature
+
+
+def load_plugin_feature(plugin_path):
+    """find an instanciate a PlugInFeature from a user-defined script"""
+    info = find_plugin_feature_script(plugin_path)
+    plugin_list = []
+    for ii in range(len(info["feature names"])):
+        ancill_info = {
+            "feature_name": info["feature names"][ii],
+            "method": info["method"],
+            "req_config": info["config required"],
+            "req_features": info["features required"],
+            "req_func": info["method check required"],
+            "priority": info["priority"],
+        }
+        plugin_list.append(PlugInFeature(info, **ancill_info))
+        # add feature label etc
+    return plugin_list
+
+
+def find_plugin_feature_script(plugin_path):
+    # find script, return info dict
+    path = pathlib.Path(plugin_path)
+    # insert the plugin directory to sys.path so we can import it
+    try:
+        plugin = importlib.import_module(path.stem)
+    except ModuleNotFoundError:
+        sys.path.insert(-1, str(path.parent))
+        plugin = importlib.import_module(path.stem)
+        # undo our path insertion
+        sys.path.pop(0)
+
```

Comment:
I think it might make sense to add `plugin_path` to `plugin.info`. Could be useful for debugging.

Reply:
Btw, I took this to mean let's add `plugin_path` as a `PlugInFeature` class attribute. I think this is useful because you can be sure where it is being loaded from.

Did you mean to add it to the `info` dict when writing the plugin script? Example:

```python
info = {
    ...
    "priority": 1,
    "plugin_path": 'path/to/file',
    "version": "0.1.0",
}
```
I can imagine this being defunc quite quickly as the user may move the plugin script around/onto a different computer etc.

Reply:
No, "plugin_path" should not be specified by the user - dclab should define it. Maybe it makes sense to have it as an attribute as opposed to in the `plugin_info` dictionary (because this might be confusing).

Reply:
Great I was on the right track. And yes it is its own attribute.